node/src/blockchain/blockchain_bridge.rs

```diff
@@ -213,60 +211,35 @@ impl BlockchainBridge {
-            match blockchain_service_url {
-                Some(url) => match Http::new(&url) {
-                    Ok((event_loop_handle, transport)) => {
-                        Self::construct_blockchain_interface(event_loop_handle, transport, chain)
-                    }
-                    Err(e) => panic!("Invalid blockchain node URL: {:?}", e),
-                },
-                None => Box::new(BlockchainInterfaceClandestine::new(chain)),
-            }
-        };
+    pub fn initialize_persistent_configuration(
+        data_directory: &Path,
+    ) -> Box<dyn PersistentConfiguration> {
         let config_dao = Box::new(ConfigDaoReal::new(
             DbInitializerReal::default()
-                .initialize(
-                    &data_directory,
-                    DbInitializationConfig::panic_on_migration(),
-                )
-                .unwrap_or_else(|err| db_connection_launch_panic(err, &data_directory)),
+                .initialize(data_directory, DbInitializationConfig::panic_on_migration())
+                .unwrap_or_else(|err| db_connection_launch_panic(err, data_directory)),
         ));
-        (
-            blockchain_interface,
-            Box::new(PersistentConfigurationReal::new(config_dao)),
-        )
+        Box::new(PersistentConfigurationReal::new(config_dao))
     }
 
-    // TODO when we have multiple chains of fundamentally different architectures and also ability to switch them,
-    // this should probably be replaced by a HashMap of a distinct blockchain interface for each chain
-    fn construct_blockchain_interface(
-        http_event_loop_handle: EventLoopHandle,
-        transport: Http,
+    pub fn initialize_blockchain_interface(
+        blockchain_service_url_opt: Option<String>,
         chain: Chain,
     ) -> Box<dyn BlockchainInterface> {
-        let gas_limit_const_part = web3_gas_limit_const_part(chain);
-        Box::new(BlockchainInterfaceNonClandestine::new(
-            transport,
-            http_event_loop_handle,
-            chain,
-            gas_limit_const_part,
-        ))
+        match blockchain_service_url_opt {
+            Some(url) => {
+                let initializer = BlockchainInterfaceInitializer {};
+                initializer.initialize_chain_compatible_interface(&url, chain)
+            }
+            None => Box::new(BlockchainInterfaceClandestine::new(chain)), //TODO make sure this is tested for BlockchainInterfaceNull (after the merge??)
+        }
     }
 
     pub fn make_subs_from(addr: &Addr<BlockchainBridge>) -> BlockchainBridgeSubs {
         BlockchainBridgeSubs {
             bind: recipient!(addr, BindMessage),
             outbound_payments_instructions: recipient!(addr, OutboundPaymentsInstructions),
-            initial_payable_payment_setup_msg: recipient!(addr, PayablePaymentsSetupMsgPayload),
+            qualified_paybles_message: recipient!(addr, QualifiedPayablesMessage),
```

Comment:
"Paybles" is misspelled. Also, does it sound right to have a recipient whose name ends with "message?" I know the previous version did too, but I think maybe it was wrong as well.